### PR TITLE
Allow ORM case insensitive string filtering

### PR DESCRIPTION
**Pull request by magikid**
I'm also taking over #165 for @adrienbrault.  Is there anything else that needs to be done for this PR?

### CODE REVIEW

File: tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php [hunk 1]
```diff
@@ -694,7 +695,43 @@ public function shouldNotFilterWithEmptyParametersAndDefaults(): void
...
+        $defaultFilterFields = 'a.title';
+        $view = $p->paginate($query, 1, 10, compact('defaultFilterFields'));
+        $items = $view->getItems();
+        $this->assertEquals(1, count($items));
+        $this->assertEquals('summer', $items[0]->getTitle());
+
+        $executed = $this->queryAnalyzer->getExecutedQueries();
+
+        // Different aliases separators according to Doctrine version
+        if (version_compare(\Doctrine\ORM\Version::VERSION, '2.5', '<')) {
```
Comment (garak) [inline tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php:727]: Not needed anymore, we don't support Doctrine before 2.7

File: tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php [hunk 2]
```diff
@@ -694,7 +695,43 @@ public function shouldNotFilterWithEmptyParametersAndDefaults(): void
         $this->assertEquals('SELECT a0_.id AS id_0, a0_.title AS title_1, a0_.enabled AS enabled_2 FROM Article a0_ LIMIT 10', $executed[5]);
     }
 
-    protected function getUsedEntityFixtures(): array
+    /**
+     * @test
+     */
+    public function shouldFilterCaseInsensitiveWhenAsked()
```
Comment (garak) [inline tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php:701]: Can we get a return type here?

File: tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php [hunk 3]
```diff
@@ -694,7 +695,43 @@ public function shouldNotFilterWithEmptyParametersAndDefaults(): void
...
+
+        // Different aliases separators according to Doctrine version
+        if (version_compare(\Doctrine\ORM\Version::VERSION, '2.5', '<')) {
+            $this->assertEquals('SELECT a0_.id AS id0, a0_.title AS title1, a0_.enabled AS enabled2 FROM Article a0_ WHERE LOWER(a0_.title) LIKE \'summer\' LIMIT 10', $executed[1]);
+        } else {
+            $this->assertEquals('SELECT a0_.id AS id_0, a0_.title AS title_1, a0_.enabled AS enabled_2 FROM Article a0_ WHERE LOWER(a0_.title) LIKE \'summer\' LIMIT 10', $executed[1]);
+        }
+    }
+
+    protected function getUsedEntityFixtures()
```
Comment (garak) [inline tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php:734]: Same as above: missing return type

**Review by garak — CHANGES_REQUESTED** [review]

File: tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php (hunk 1 above)
Comment (magikid) [reply tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php]: I've removed the version check.

File: tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php (hunk 2 above)
Comment (magikid) [reply tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php]: Done, I've added `void`

File: tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php (hunk 3 above)
Comment (magikid) [reply tests/Test/Pager/Subscriber/Filtration/Doctrine/ORM/QueryTest.php]: Done.  I've added a native PHP return type of `array` and a PHPDoc comment with the return type of `Article[]`.

**Review by garak — APPROVED** [review]

**Comment by garak** [comment]
Can you fix conflicts?

**Comment by magikid** [comment]
I rebased this branch onto master to pick up the changes there.

**Pull request merged**